server/mqtt.go

```diff
@@ -1439,26 +1439,32 @@ func (s *Server) mqttCreateAccountSessionManager(acc *Account, quitCh chan struc
 		}
 	}
 
+	// Delete the old (legacy) consumer, from v2.10.10 and before.
+	rmLegacyDurName := mqttRetainedMsgsStreamName + "_" + jsa.id
+	if _, err := jsa.deleteConsumer(mqttRetainedMsgsStreamName, rmLegacyDurName); isErrorOtherThan(err, JSConsumerNotFoundErr) {
```

Comment:
I don't think this should bubble up error.

Reply:
2/5 we shouldn't continue initializing MQTT - what else would we return?

Reply:
Should not matter if we get an error trying to delete legacy consumer IMO. Should not prevent us from proceeding.